src/pip/_internal/commands/cache.py

```diff
@@ -0,0 +1,152 @@
+        %prog list [name]
+        %prog remove <pattern>
+        %prog purge
+    """
+
+    def __init__(self, *args, **kw):
+        # type: (*Any, **Any) -> None
+        super(CacheCommand, self).__init__(*args, **kw)
+
+    def run(self, options, args):
+        # type: (Values, List[Any]) -> int
+        handlers = {
+            "info": self.get_cache_info,
+            "list": self.list_cache_items,
+            "remove": self.remove_cache_items,
+            "purge": self.purge_cache,
+        }
+
+        # Determine action
+        if not args or args[0] not in handlers:
+            logger.error("Need an action ({}) to perform.".format(
+                ", ".join(sorted(handlers)))
+            )
+            return ERROR
+
+        action = args[0]
+
+        # Error handling happens here, not in the action-handlers.
+        try:
+            handlers[action](options, args[1:])
+        except PipError as e:
+            logger.error(e.args[0])
+            return ERROR
+
+        return SUCCESS
+
+    def get_cache_info(self, options, args):
+        # type: (Values, List[Any]) -> None
+        num_packages = len(self._find_wheels(options, '*'))
+
+        cache_location = self._wheels_cache_dir(options)
+        cache_size = filesystem.friendly_directory_size(cache_location)
+
+        message = textwrap.dedent("""
+            Cache info:
+              Location: {location}
+              Packages: {package_count}
+              Size:     {size}
+        """).format(
+            location=cache_location,
+            package_count=num_packages,
+            size=cache_size,
+        ).strip()
+
+        logger.info(message)
+
+    def list_cache_items(self, options, args):
+        # type: (Values, List[Any]) -> None
+        if len(args) > 1:
+            raise CommandError('Too many arguments')
+
+        if args:
+            pattern = args[0]
+        else:
+            pattern = '*'
+
+        files = self._find_wheels(options, pattern)
+
+        if not files:
+            logger.info('Nothing is currently cached.')
+            return
+
+        results = []
+        for filename in files:
+            wheel = os.path.basename(filename)
+            size = filesystem.friendly_file_size(filename)
+            results.append(' - {} ({})'.format(wheel, size))
+        logger.info('Current cache contents:\n')
+        logger.info('\n'.join(sorted(results)))
+
+    def remove_cache_items(self, options, args):
+        # type: (Values, List[Any]) -> None
+        if len(args) > 1:
+            raise CommandError('Too many arguments')
+
+        if not args:
+            raise CommandError('Please provide a pattern')
+
+        files = self._find_wheels(options, args[0])
+        if not files:
+            raise CommandError('No matching packages')
+
+        for filename in files:
+            os.unlink(filename)
+            logger.debug('Removed %s', filename)
+        logger.info('Removed %s files', len(files))
```

Comment:
Perhaps also deal with singular case? So:
```
Removed 1 file
Removed 2 files
```
Instead of:
```
Removed 1 files
Removed 2 files
```

Reply:
Alternatively, "file(s)".

Reply:
Computers are bad at many things, but one thing they're really good at is counting :)

Reply:
Personally I prefer file(s). I am probably super biased since my native language does not have plurals 😛  but IMO locale-sensitive string formatting is bad for logging in general.

Reply:
Changing it to `file(s)`.

Reply:
I'd like to suggest not to use "file(s)".

If it absolutely must be static for logging, may I suggest rewriting like this?

```diff
-        logger.info('Removed %s files', len(files))
+        logger.info('Files removed: %s', len(files))
```

Here are some style guides that say to avoid "(s)":

* https://developers.google.com/style/plurals-parentheses

![image](https://user-images.githubusercontent.com/1324225/72334880-b8cba600-36c6-11ea-8278-5262ab55c3ad.png)


* IBM Style Guide:
![image](https://user-images.githubusercontent.com/1324225/72334587-3642e680-36c6-11ea-82b5-318978196acb.png)

* Global English Style Guide: Writing Clear, Translatable Documentation for a Global Market:

![image](https://user-images.githubusercontent.com/1324225/72334625-48bd2000-36c6-11ea-98b5-33f5b03ec670.png)

Reply:
*Files removed* works for me.

Reply:
@hugovk That works for me.

Reply:
@hugovk I appreciate your feedback here. I'm changing it to `Files removed: %s` like you suggested. :+1: